linode/networkreservedips/framework_datasource_schema.go

```diff
@@ -0,0 +1,91 @@
+package networkreservedips
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type ReservedIPObject struct {
+	ID         types.String `tfsdk:"id"`
+	Address    types.String `tfsdk:"address"`
+	Region     types.String `tfsdk:"region"`
+	Gateway    types.String `tfsdk:"gateway"`
+	SubnetMask types.String `tfsdk:"subnet_mask"`
+	Prefix     types.Int64  `tfsdk:"prefix"`
+	Type       types.String `tfsdk:"type"`
+	Public     types.Bool   `tfsdk:"public"`
+	RDNS       types.String `tfsdk:"rdns"`
+	LinodeID   types.Int64  `tfsdk:"linode_id"`
+	Reserved   types.Bool   `tfsdk:"reserved"`
+}
+
+var reservedIPObjectType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"id":          types.StringType,
+		"address":     types.StringType,
+		"region":      types.StringType,
+		"gateway":     types.StringType,
+		"subnet_mask": types.StringType,
+		"prefix":      types.Int64Type,
+		"type":        types.StringType,
+		"public":      types.BoolType,
+		"rdns":        types.StringType,
+		"linode_id":   types.Int64Type,
+		"reserved":    types.BoolType,
+	},
+}
+
+var frameworkDataSourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"region": schema.StringAttribute{
+			Description: "The Region in which to reserve the IP address.",
+			Optional:    true,
+		},
+		"address": schema.StringAttribute{
+			Description: "The reserved IP address.",
+			Optional:    true,
+		},
```

Comment:
I have made it optional since I am implementing both fetch and list endpoints in the same datasource. But now it seems like it is indeed a better idea to separate the functionalities into individual datasources. I will be doing that in the next commit.